plugins/org.jkiss.dbeaver.ext.vertica/src/org/jkiss/dbeaver/ext/vertica/model/QueryTransformerLimitVertica.java

```diff
@@ -36,14 +34,10 @@ public QueryTransformerLimitVertica() {
     @Override
     public boolean isApplicableTo(SQLQuery query) {
         Statement statement = query.getStatement();
-        return statement != null && isLimitApplicable(statement);
-    }
-
-    public boolean isLimitApplicable(Statement statement) {
-        if (statement instanceof PlainSelect plainSelect) {
-            return plainSelect.getFromItem() != null && CommonUtils.isEmpty(plainSelect.getIntoTables())
-                && plainSelect.getLimit() == null && plainSelect.getTop() == null && plainSelect.getForUpdateTable() == null;
+        if (statement instanceof PlainSelect select) {
+            return query.isPlainSelect() && select.getFromItem() != null;
```

Comment:
Why `&& CommonUtils.isEmpty(plainSelect.getIntoTables()) && plainSelect.getLimit() == null && plainSelect.getTop() == null` is deleted?

Reply:
> Why `&& CommonUtils.isEmpty(plainSelect.getIntoTables()) && plainSelect.getLimit() == null && plainSelect.getTop() == null` is deleted?

it duplicates query.isPlainSelect()